bizflycloud/resource_bizflycloud_server.go

```diff
@@ -201,10 +76,48 @@ func resourceBizflyCloudServerCreate(d *schema.ResourceData, meta interface{}) e
 		Password:         d.Get("password").(bool),
 		RootDisk:         &rootDiskPayload,
 		NetworkPlan:      d.Get("network_plan").(string),
-		VPCNetworkIds:    readStringArray(d.Get("vpc_network_ids").(*schema.Set).List()),
 		BillingPlan:      d.Get("billing_plan").(string),
 		UserData:         d.Get("user_data").(string),
 	}
+	var (
+		isCreatedWan         bool
+		usingV6Wan           bool
+		freeWanV4BlockCount  int
+		freeWanV6BlockCount  int
+		networkInterfaceIDs  []string
+		freeWanV4FirewallIDs []string
+		freeWanV6FirewallIDs []string
+	)
+	attachPortIPFirewallIDs := make(map[string][]string)
+	for _, v := range d.Get("free_wan").(*schema.Set).List() {
+		freeWan := v.(map[string]interface{})
+		if freeWan["ip_version"].(int) == 4 {
+			freeWanV4BlockCount++
+			if freeWanV4BlockCount > 1 {
+				return fmt.Errorf("only one free WAN V4 is allowed")
+			}
+			isCreatedWan = true
+			freeWanV4FirewallIDs = readStringArray(freeWan["firewall_ids"].(*schema.Set).List())
+		} else {
+			freeWanV6BlockCount++
+			usingV6Wan = true
+			if freeWanV6BlockCount > 1 {
+				return fmt.Errorf("only one free WAN V6 is allowed")
```

Comment:
I think we can break free_wan into two parameters one for IPv4 and one for IPv6.

Reply:
Do you mean the schema would become 
```
free_wan_v4
{
}

free_wan_v6 {
}
```

Reply:
Yes. And I think we should use other terminology instead of free_wan such as default_public_ipv4 (default_public_ipv6 ). Just a suggestion.